### PR TITLE
chore: bundle license to release artifacts

### LINKED ISSUE #162: Licence compliance (automated if possible)
Fossa has identified [license issues](https://app.fossa.com/projects/git%2Bgithub.com%2Fopen-feature%2Fgo-sdk-contrib/refs/branch/main/8ee4c520f9efdf42bc2dc715523adc0b1cf7d76b/issues/licensing?page=1&count=20&sort=created_at_desc&status=active&filter%5Btype%5D%5B0%5D=policy_flag&revisionScanId=38217278) with this repo. The only requirement I can see is a requirement to include licenses.

Most are Mozilla. We need to add declarations for these. It seems like we may be able to automate this using: https://github.com/google/go-licenses. We should do this within each module.

### PR DESCRIPTION
**Pull request by Kavindu-Dodan**
## This PR

Fixes #162 by bundling individual artifact's third party licenses into an archive in releases.

Archive naming -  `third-party-license.tar.gz`

Additionally, license generation errors are accumulated at `license-errors.txt` for any interested party to check.

### CODE REVIEW

**Review by beeme1mr — APPROVED** [review]

File: .github/workflows/release-please.yaml [hunk 1]
```diff
@@ -21,24 +21,47 @@ jobs:
...
         uses: actions/checkout@v3
+      # Bundle SBOM
       - name: Generate SBOM
         uses: CycloneDX/gh-gomod-generate-sbom@v1
         with:
           version: v1
           args: mod -licenses -json -output bom.json ${{ matrix.release }}
       - name: Upload SBOM to release
-        uses: svenstaro/upload-release-action@v2
+        uses: softprops/action-gh-release@de2c0eb89ae2a093876385947365aca7b0e5f844
```
Comment (thisthat) [inline .github/workflows/release-please.yaml:43]: why changing the action?

Comment (Kavindu-Dodan) [reply .github/workflows/release-please.yaml:43]: Good point, I switched to `softprops/action-gh-release` as I have seen the usage in other repositories I maintain. But, seems `svenstaro/upload-release-action` fits well here. 

I will made the swithch

Comment (Kavindu-Dodan) [reply .github/workflows/release-please.yaml:43]: One weakness of `svenstaro/upload-release-action` is that it lacks multi-file upload capability. I will refactor the code to use `svenstaro/upload-release-action` in a single job step

**Review by thisthat — APPROVED** [review]
LGTM 👍  Thanks @Kavindu-Dodan

**Pull request merged**